src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsEmulatorConfigWriterHook.cs

```diff
@@ -0,0 +1,67 @@
+        {
+            // No-op if there is no Azure Event Hubs emulator resource.
+            return Task.CompletedTask;
+        }
+
+        var configFileMount = emulatorResource.Annotations.OfType<ContainerMountAnnotation>().Single(v => v.Target == AzureEventHubsEmulatorResource.EmulatorConfigJsonPath);
+
+        if (!emulatorResource.Hubs.Any())
+        {
+            // No-op if there are no Azure Event Hubs resources present.
+            return Task.CompletedTask;
+        }
+
+        using var stream = new FileStream(configFileMount.Source!, FileMode.Create);
+        using var writer = new Utf8JsonWriter(stream);
+
+        writer.WriteStartObject();                      // {
+        writer.WriteStartObject("UserConfig");          //   "UserConfig": {
+        writer.WriteStartArray("NamespaceConfig");      //     "NamespaceConfig": [
+        writer.WriteStartObject();                      //       {
+        writer.WriteString("Type", "EventHub");         //         "Type": "EventHub",
+        writer.WriteString("Name", "emulatorNs1");      //         "Name": "emulatorNs1"
+        writer.WriteStartArray("Entities");             //         "Entities": [
+
+        foreach (var hub in emulatorResource.Hubs)
+        {
+            // The default consumer group ('$default') is automatically created, and this
+            // is also the one used by a consumer client if no ConsumerGroup value
+            // is provided.
+            writer.WriteStartObject();                  //           {
+            writer.WriteString("Name", hub.Name);       //             "Name": "hub",
+            writer.WriteString("PartitionCount", "2");  //             "PartitionCount": "2",
+            writer.WriteStartArray("ConsumerGroups");   //              "ConsumerGroups": [
+            writer.WriteStartObject();                  //                {
+            writer.WriteString("Name", "cg1");          //                  "Name": "cg1"
+            writer.WriteEndObject();                    //                }
+            writer.WriteEndArray();                     //              ]
+            writer.WriteEndObject();                    //            }
+        }
+
+        writer.WriteEndArray();                         //          ] (/Entities)
+        writer.WriteEndObject();                        //        } 
+        writer.WriteEndArray();                         //      ], (/NamespaceConfig)
+        writer.WriteStartObject("LoggingConfig");       //      "LoggingConfig": {
+        writer.WriteString("Type", "File");             //        "Type": "File"
+        writer.WriteEndObject();                        //        } (/LoggingConfig)
+
+        writer.WriteEndObject();                        //        } (/UserConfig)
+        writer.WriteEndObject();                        //      } (/Root)
```

Comment:
(nit) the nesting of `}` in the comments is off

Reply:
Will fix.

PS: The `Utf8JsonWriter` is almost overkill here. A template might feel nicer, but I still preferred this construct to something like `""" ... """ + loop + """ ... """`

Reply:
> The Utf8JsonWriter is almost overkill here. 

I had that comment written up... and then deleted it. 😄

Reply:
Done